### PR TITLE
Accept Hspec command-line arguments

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# autospec
+
+In it's simplest form, you run `autospec` with the `Main` module of your test
+suite as an argument:
+
+    autospec test/Spec.hs
+
+Note that `autospec` picks up options from `.ghci`-files.  You can provided
```
Comment (soenkehahn) [inline README.md]: `provided` -> `provide`

File: README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+additional GHC options on the command lines:
```
Comment (soenkehahn) [inline README.md]: `lines` -> `line`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    autospec -isrc -itest test/Spec.hs
+
+Command-line arguments that looks like Hspec options are passed to Hspec.  To
```
Comment (soenkehahn) [inline README.md]: `looks` -> `look`

File: README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+avoid ambiguity, GHC options have to be given before any Hspec options:
+
+    autospec -isrc -itest test/Spec.hs --no-color --match foo
+
+All command-line arguments after the last `--` are passed to Hspec, regardless
+how the look:
```
Comment (soenkehahn) [inline README.md]: `the` -> `they`?

File: README.md (hunk 1 above)
Comment (sol) [inline README.md]: Good catch :)

File: README.md (hunk 4 above)
Comment (sol) [inline README.md]: looks like I'm tired.....

thx

File: src/Options.hs [hunk 5]
```diff
@@ -0,0 +1,38 @@
+module Options (splitArgs) where
+
+import           Data.List
+import           System.Console.GetOpt
+
+splitArgs :: [String] -> ([String], [String])
+splitArgs args = case break (== "--") $ reverse args of
+  (xs, "--" : ys) -> (reverse ys, reverse xs)
+  _ -> case filter isHspecArgs $ tails args of
+    x:_ -> (dropEnd (length x) args, x)
```
Comment (soenkehahn) [inline src/Options.hs]: Spaces around `:`?

File: src/Options.hs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  , Option []  ["depth"]            (ReqArg (const ()) "N")
+  , Option "a" ["qc-max-success"]   (ReqArg (const ()) "N")
+  , Option ""  ["qc-max-size"]      (ReqArg (const ()) "N")
+  , Option ""  ["qc-max-discard"]   (ReqArg (const ()) "N")
+  , Option []  ["seed"]             (ReqArg (const ()) "N")
+  , Option []  ["print-cpu-time"]   (NoArg ())
+  , Option []  ["dry-run"]          (NoArg ())
+  , Option []  ["fail-fast"]        (NoArg ())
+  , Option "r" ["rerun"]            (NoArg ())
+  ]
```
Comment (soenkehahn) [inline src/Options.hs]: If you're using string literals for e.g. `"m"` I would also use string literals for `""`. I would prefer `['m']` though as this is not a `String` but a list of `Char`s.

File: src/Interpreter.hs [hunk 7]
```diff
@@ -7,16 +8,28 @@ module Interpreter (
...
+}
+
+new :: [String] -> IO Session
 new args = do
-  ghci <- GhciWrapper.new defaultConfig{configVerbose = True, configIgnoreDotGhci = False} args
+  let (ghciArgs, hspecArgs) = splitArgs args
+  ghci <- GhciWrapper.new defaultConfig{configVerbose = True, configIgnoreDotGhci = False} ghciArgs
   _ <- eval ghci (":set prompt " ++ show "")
-  return ghci
+  _ <- eval ghci ("import qualified System.Environment" ++ show "")
```
Comment (soenkehahn) [inline src/Interpreter.hs]: What's the `show ""` about?

**Comment by soenkehahn** [comment]
Apart from comments LGTM.

File: src/Interpreter.hs (hunk 7 above)
Comment (sol) [inline src/Interpreter.hs]: copy & paste, thanks for catching :)

File: src/Options.hs (hunk 6 above)
Comment (sol) [inline src/Options.hs]: You have a point, but I just tried I think the current form is easiest to read.  I use empty list everywhere now, though.  Eventually this should be exported and re-used from Hspec.

File: src/Options.hs (hunk 5 above)
Comment (sol) [inline src/Options.hs]: addressed ;)

Comment (sol) [inline src/Options.hs]: But I think I do it like this most of he time.

Comment (sol) [inline src/Options.hs]: Not that there is any good reason for it ...

**Pull request merged**